Autopilot/PathManager/Src/cruisingState.cpp

```diff
@@ -0,0 +1,349 @@
+/*
```

Comment:
I really love the looks of this file. I trust that the unit tests confirm it does what you intend, but it's really good to see something complicated done in a way that's fairly readable.